openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingPopover.tsx

```diff
@@ -1,133 +1,90 @@
+  // Duplicate
   const [duplicate, setDuplicate] = useState(false);
-
-  // Fetching data
-  const { updateInjectResultDto } = useContext<InjectResultDtoContextType>(InjectResultDtoContext);
-  const { teams } = useHelper((helper: TeamsHelper) => ({
-    teams: helper.getTeams(),
-  }));
-  useDataLoader(() => {
-    dispatch(fetchTeams());
-  });
-
-  // UPDATE
-  const submitUpdateAtomicTesting = async (data: Inject) => {
-    const toUpdate = R.pipe(
-      R.pick([
-        'inject_tags',
-        'inject_title',
-        'inject_type',
-        'inject_injector_contract',
-        'inject_description',
-        'inject_content',
-        'inject_all_teams',
-        'inject_documents',
-        'inject_assets',
-        'inject_asset_groups',
-        'inject_teams',
-        'inject_tags',
-      ]),
-    )(data);
-    updateAtomicTesting(atomic.inject_id, toUpdate).then((result: { data: InjectResultDTO }) => {
-      updateInjectResultDto(result.data);
-      if (setOpenEdit) {
-        setOpenEdit(null);
-      }
-    });
-  };
-
-  // DUPLICATE
+  const handleOpenDuplicate = () => setDuplicate(true);
+  const handleCloseDuplicate = () => setDuplicate(false);
   const submitDuplicate = async () => {
     await duplicateAtomicTesting(atomic.inject_id).then((result: { data: InjectResultDTO }) => {
+      handleCloseDuplicate();
       navigate(`/admin/atomic_testings/${result.data.inject_id}`);
     });
-    setDuplicate(false);
   };
 
-  // DELETE
+  // Edition
+  const [edition, setEdition] = useState(false);
+  const handleOpenEdit = () => setEdition(true);
+  const handleCloseEdit = () => setEdition(false);
+
+  // Deletion
+  const [deletion, setDeletion] = useState(false);
+  const handleOpenDelete = () => setDeletion(true);
+  const handleCloseDelete = () => setDeletion(false);
   const submitDelete = () => {
     deleteAtomicTesting(atomic.inject_id).then(() => {
-      setDeletion(false);
-      navigate('/admin/atomic_testings');
+      handleCloseDelete();
+      if (onDelete) onDelete(atomic.inject_id);
     });
   };
 
-  const entries: PopoverEntry[] = [];
-  if (atomic.inject_injector_contract !== null) {
-    if (actions.includes('Duplicate')) {
-      entries.push({ label: 'Duplicate', action: () => setDuplicate(true) });
-    }
-    if (actions.includes('Update')) {
-      entries.push({ label: 'Update', action: () => (isNotEmptyField(setOpenEdit) ? setOpenEdit(atomic.inject_id) : setEdition(true)) });
-    }
-  }
-  if (actions.includes('Delete')) entries.push({ label: 'Delete', action: () => setDeletion(true) });
+  // Button Popover
+  const entries = [];
+  if (actions.includes('Duplicate')) entries.push({ label: 'Duplicate', action: () => handleOpenDuplicate() });
+  if (actions.includes('Update')) entries.push({ label: 'Update', action: () => handleOpenEdit() });
+  if (actions.includes('Delete')) entries.push({ label: 'Delete', action: () => handleOpenDelete() });
```

Comment:
Or here, but we need to filter update and duplicate when the inject doesn't have a contract.